File: ruby/lib/steam/community/tf2/tf2_engineer.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/community/tf2/tf2_class'

# Represents the stats for the Team Fortress 2 Engineer class for a specific
# user
#
# @author Sebastian Staudt
class TF2Engineer

  include TF2Class

  # Returns the maximum number of buildings built by the player in a single
  # life as an Engineer
  #
  # @return [Fixnum] Maximum number of buildings built
  attr_reader :max_buildings_built

  # Returns the maximum number of enemies killed by sentry guns built by the
  # player in a single life as an Engineer
  #
  # @return [Fixnum] Maximum number of sentry kills
  attr_reader :max_sentry_kills

  # Returns the maximum number of teammates teleported by teleporters built by
  # the player in a single life as an Engineer
  #
  # @return [Fixnum] Maximum number of teleports
  attr_reader :max_teleports

  # Creates a new instance of the Engineer class based on the given XML data
  #
  # @param [REXML::Element] class_data The XML data for this Engineer
  def initialize(class_data)
    super class_data

    @max_buildings_built = class_data.elements['ibuildingsbuilt'].text.to_i
    @max_teleports       = class_data.elements['inumteleports'].text.to_i
    @max_sentry_kills    = class_data.elements['isentrykills'].text.to_i
  end

end

File: java/src/main/java/com/github/koraktor/steamcondenser/steam/community/tf2/TF2Engineer.java
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2011, Sebastian Staudt
 */

package com.github.koraktor.steamcondenser.steam.community.tf2;

import org.w3c.dom.Element;

/**
 * Represents the stats for the Team Fortress 2 Engineer class for a specific
 * user
 *
 * @author     Sebastian Staudt
 */
public class TF2Engineer extends TF2Class
{
    private int maxBuildingsBuilt;

    private int maxSentryKills;

    private int maxTeleports;

    /**
     * Creates a new instance of the Engineer class based on the given XML data
     *
     * @param classData The XML data for this Engineer
     */
    public TF2Engineer(Element classData)
    {
        super(classData);

        this.maxBuildingsBuilt = Integer.parseInt(classData.getElementsByTagName("ibuildingsbuilt").item(0).getTextContent());
        this.maxSentryKills    = Integer.parseInt(classData.getElementsByTagName("isentrykills").item(0).getTextContent());
        this.maxTeleports      = Integer.parseInt(classData.getElementsByTagName("inumteleports").item(0).getTextContent());
    }

    /**
     * Returns the maximum number of buildings built by the player in a single
     * life as an Engineer
     *
     * @return Maximum number of buildings built
     */
    public int getMaxBuildingsBuilt()
    {
        return this.maxBuildingsBuilt;
    }

    /**
     * Returns the maximum number of enemies killed by sentry guns built by the
     * player in a single life as an Engineer
     *
     * @return Maximum number of sentry kills
     */
    public int getMaxSentryKills()
    {
        return this.maxSentryKills;
    }

    /**
     * Returns the maximum number of teammates teleported by teleporters built
     * by the player in a single life as an Engineer
     *
     * @return Maximum number of teleports
     */
    public int getMaxTeleports()
    {
        return this.maxTeleports;
    }

}
